src/lib.rs

```diff
@@ -15,9 +16,22 @@ use rustc_hash::FxHashMap as HashMap;
 
 type Rank = u32;
 
+const LARGE_ENCODER_CHARACTER_LIMIT: usize = 500;
```

Comment:
in the Java version this was controlled by an environmental variable, which enabled us to run all tests against both implementations - should I do it here as well?